pkg/agent/flowexporter/connections/connections.go

```diff
@@ -0,0 +1,149 @@
+		connTrack:   connTrack,
+		ifaceStore:  ifaceStore,
+	}
+}
+
+// Run polls the connTrack module periodically to get connections. These connections are used
+// to build connection store.
+func (cs *connectionStore) Run(stopCh <-chan struct{}) {
+	klog.Infof("Starting conntrack polling")
+	for {
+		select {
+		case <-stopCh:
+			break
+		case <-time.After(flowexporter.PollInterval):
+			_, err := cs.poll()
+			if err != nil {
+				klog.Errorf("Error during conntrack poll cycle: %v", err)
+			}
+		}
+	}
+}
+
+// addOrUpdateConn updates the connection if it is already present, i.e., update timestamp, counters etc.,
+// or add a new Connection by 5-tuple of the flow along with local Pod and PodNameSpace.
+func (cs *connectionStore) addOrUpdateConn(conn *flowexporter.Connection) {
+	connKey := flowexporter.NewConnectionKey(conn)
+
+	existingConn, exists := cs.getConnByKey(connKey)
+
+	cs.mutex.Lock()
+	defer cs.mutex.Unlock()
+	if exists {
+		// Update the necessary fields that are used in generating flow records.
+		// Can same 5-tuple flow get deleted and added to conntrack table? If so use ID.
+		existingConn.StopTime = conn.StopTime
+		existingConn.OriginalBytes = conn.OriginalBytes
+		existingConn.OriginalPackets = conn.OriginalPackets
+		existingConn.ReverseBytes = conn.ReverseBytes
+		existingConn.ReversePackets = conn.ReversePackets
+		// Reassign the flow to update the map
+		cs.connections[connKey] = *existingConn
+		klog.V(2).Infof("Antrea flow updated: %v", existingConn)
+	} else {
+		var srcFound, dstFound bool
+		sIface, srcFound := cs.ifaceStore.GetInterfaceByIP(conn.TupleOrig.SourceAddress.String())
+		dIface, dstFound := cs.ifaceStore.GetInterfaceByIP(conn.TupleReply.SourceAddress.String())
+		if !srcFound && !dstFound {
+			klog.Warningf("Cannot map any of the IP %s or %s to a local Pod", conn.TupleOrig.SourceAddress.String(), conn.TupleReply.SourceAddress.String())
+		}
+		if srcFound && sIface.Type == interfacestore.ContainerInterface {
+			conn.SourcePodName = sIface.ContainerInterfaceConfig.PodName
+			conn.SourcePodNamespace = sIface.ContainerInterfaceConfig.PodNamespace
+		}
+		if dstFound && dIface.Type == interfacestore.ContainerInterface {
+			conn.DestinationPodName = dIface.ContainerInterfaceConfig.PodName
+			conn.DestinationPodNamespace = dIface.ContainerInterfaceConfig.PodNamespace
+		}
+		klog.V(2).Infof("New Antrea flow added: %v", conn)
+		// Add new antrea connection to connection store
+		cs.connections[connKey] = *conn
+	}
+}
+
+func (cs *connectionStore) getConnByKey(flowTuple flowexporter.ConnectionKey) (*flowexporter.Connection, bool) {
+	cs.mutex.Lock()
+	defer cs.mutex.Unlock()
+	conn, found := cs.connections[flowTuple]
+	return &conn, found
+}
+
+func (cs *connectionStore) IterateCxnMapWithCB(updateCallback flowexporter.FlowRecordUpdate) error {
+	cs.mutex.Lock()
+	defer cs.mutex.Unlock()
+
+	for k, v := range cs.connections {
+		cs.mutex.Unlock()
+		err := updateCallback(k, v)
```

Comment:
Do you think we could enable a mode that send only changed flows/connections?

Reply:
Added comment in L#120 about the related optimization. Yes it is possible, if stats are not changing, we can skip flow records. If a connection is closed and re-established again with the same 5-tuple in the same window before sending next export, we have to consider that as a new connection. We could use conntrack ID to differentiate both connections or start time stamp. Any suggestions?